lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,384 @@
+          # green. (They're yellow under vagrant.)
+          sh.echo 'R for Travis-CI is not officially supported, ' +
+                  'but is community maintained.', ansi: :green
+          sh.echo 'Please file any issues using the following link',
+                  ansi: :green
+          sh.echo '  https://github.com/travis-ci/travis-ci/issues' +
+                  '/new?labels=community:r', ansi: :green
+          sh.echo 'and mention \`@craigcitro\`, \`@eddelbuettel\` and ' +
+                  '\`@hadley\` in the issue', ansi: :green
+
+          # TODO(craigcitro): python-software-properties?
+          sh.echo 'Installing R'
+          case config[:os]
+          when 'linux'
+            # Set up our CRAN mirror.
+            sh.cmd 'sudo add-apt-repository ' +
+                   "\"deb #{config[:cran]}/bin/linux/ubuntu " +
+                   "$(lsb_release -cs)/\""
+            sh.cmd 'sudo apt-key adv --keyserver keyserver.ubuntu.com ' +
+                   '--recv-keys E084DAB9'
+
+            # Add marutter's c2d4u repository.
+            sh.cmd 'sudo add-apt-repository -y "ppa:marutter/rrutter"'
+            sh.cmd 'sudo add-apt-repository -y "ppa:marutter/c2d4u"'
+
+            # Update after adding all repositories. Retry several
+            # times to work around flaky connection to Launchpad PPAs.
+            sh.cmd 'sudo apt-get update -qq', retry: true
+
+            # Install an R development environment. qpdf is also needed for
+            # --as-cran checks:
+            #   https://stat.ethz.ch/pipermail/r-help//2012-September/335676.html
+            sh.cmd 'sudo apt-get install --no-install-recommends r-base-dev ' +
+                   'r-recommended qpdf', retry: true
+
+            # Change permissions for /usr/local/lib/R/site-library
+            # This should really be via 'sudo adduser travis staff'
+            # but that may affect only the next shell
+            sh.cmd 'sudo chmod 2777 /usr/local/lib/R /usr/local/lib/R/site-library'
+
+          when 'osx'
+            sh.cmd 'brew update', retry: true
+
+            # Install from latest CRAN binary build for OS X
+            sh.cmd "wget #{config[:cran]}/bin/macosx/R-latest.pkg " +
+                   '-O /tmp/R-latest.pkg'
+
+            sh.echo 'Installing OS X binary package for R'
+            sh.cmd 'sudo installer -pkg "/tmp/R-latest.pkg" -target /'
+            sh.rm '/tmp/R-latest.pkg'
+
+          else
+            sh.failure "Operating system not supported: #{config[:os]}"
+          end
+
+          setup_latex
+          
+          setup_bioc if needs_bioc?
+          setup_pandoc if config[:pandoc]
+        end
+
+        def announce
+          super
+
+          sh.cmd 'Rscript -e \'sessionInfo()\''
+          sh.echo ''
+        end
+
+        def install
+          super
+
+          # Install any declared packages
+          apt_install config[:apt_packages]
```

Comment:
yes -- we know using sudo is not desirable, and is going to prevent container use. the recommendation [we got before](https://github.com/travis-ci/travis-ci/issues/1549#issuecomment-72396184) was to start here and remove `sudo` use in the future.

the main issue is that we don't know the *complete* list of packages up front -- we install user-specified R dependencies via `apt-get` to speed up install time (cf `r_binary_install` below).